seaworthy/definitions.py

```diff
@@ -52,16 +51,41 @@ def remove(self, **kwargs):
         self.helper.remove(self.inner(), **kwargs)
         self._inner = None
 
-    def __enter__(self):
+    def setup(self, helper=None):
+        """
+        Setup this resource so that is ready to be used in a test. If the
+        resource has already been created, this call does nothing.
+
+        For most resources, this just involes creating the resource in Docker.
+
+        :param helper:
+            The resource helper to use, if one was not provided when this
+            resource definition was created.
+        """
+        if self.created:
+            return
+
+        self.set_helper(helper)
         self.create()
+
+    def teardown(self):
+        """
+        Teardown this resource so that it no longer exists in Docker. If the
+        resource has already been removed, this call does nothing.
+
+        For most resources, this just involves removing the resource in Docker.
+        """
+        if not self.created:
+            return
+
+        self.remove()
```

Comment:
Why not `if self.created: self.remove()`?

Reply:
Just me being pedantic and making it the same as setup().. but maybe that's worse..